src/main/java/com/bookshop/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package com.bookshop.security;
+
+import com.bookshop.exception.EntityNotFoundException;
+import com.bookshop.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username).orElseThrow(
+                () -> new EntityNotFoundException("There is no user with email: " + username));
```

Comment:
so `UsernameNotFoundException` or `EntityNotFoundException` ?:)

remove the `throws`